server/svix-server/tests/e2e_endpoint.rs

```diff
@@ -425,80 +425,70 @@ async fn test_recovery_should_fail_if_start_time_too_old() {
         .unwrap();
 }
 
-async fn create_failed_message_attempts() -> (
-    TestClient,
-    ApplicationId,
-    EndpointId,
-    MessageOut,
-    MessageOut,
-    usize,
-    [DateTime<Utc>; 3],
-) {
+#[tokio::test]
+async fn test_recovery_expected_retry_counts() {
     let mut cfg = get_default_test_config();
+
     cfg.retry_schedule = (0..2)
         .into_iter()
         .map(|_| Duration::from_millis(1))
         .collect();
 
-    let (client, _jh) = start_svix_server_with_cfg(&cfg);
+    // total attempts for a failed message should be 1 (first attempt) + length of retry_schedule:
+    let base_attempt_cnt = 1 + &cfg.retry_schedule.len();
 
-    let app_id = create_test_app(&client, "app1").await.unwrap().id;
+    for (i, (msg_1_retry_cnt, msg2_retry_cnt)) in [
+        // expected number of additional retry attempts for (msg1, msg2) if recover `since` is before msg 1
+        // (i.e., both messages will have one additional attempt after recovery):
+        (1, 1),
+        // expected values if recover `since` is before msg 2:
+        // (i.e., msg 2 will have one additional attempt after recovery):
+        (0, 1),
+        // expected values if recover `since` is after msg 2:
+        // (i.e., neither msg will have additional attempt after recovery):
+        (0, 0),
```

Comment:
This part is very confusing and is obfuscating what's actually going on. If it's just for filtering the returned messages, why not literally just have the same create app, endpoint, msg sequence, and then just manually three `list` calls with different since. This is just very confusing and doesn't actually save any code/effort.